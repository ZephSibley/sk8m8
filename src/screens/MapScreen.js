import React, { useState, useEffect } from 'react';
import NativeSelect from '@material-ui/core/NativeSelect';

import locate from '../utility/locate';

const MapScreen = () => {
    const [location, setLocation] = useState({
            lat: null,
            long: null,
        });
    // Marker format TBD
    const [markers, setMarkers] = useState(null);
    const [radius, setRadius] = useState(10);

    useEffect(() => {
        getLocation();
        fetchMarkers();
    }, [location, markers, radius]);

    const getLocation = async () => {
        const {latitude, longitude} = await locate();
        setLocation({
                lat: latitude, 
                long: longitude
        });
    }

    const fetchMarkers = async () => {
        // Need to send state.location and state.radius
        const data = await fetch(
            ''
        );
        //const markers = await data.json();
        setMarkers(data);
    }

    const handleRadiusChange = event => {
        setRadius(event.target.value)
        fetchMarkers();
    }

    return (
        // TODO: render map but give loading spinner for markers
        markers ? <div>loading...</div> :
        <div>
            Map Stuff
            <NativeSelect
                value={radius}
                onChange={handleRadiusChange('radius')}
                inputProps={{
                    name: 'radius',
                }}
            >
                <option value={0} />
                <option value={5} />
                <option value={10} />
                <option value={25} />
                <option value={50} />
                <option value={100} />
            </NativeSelect>
        </div>
    )
}

export default MapScreen;