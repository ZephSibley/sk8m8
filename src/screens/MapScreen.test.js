import React from 'react';
import renderer from 'react-test-renderer';
import MapScreen from './MapScreen';

test('Basic Rendering', () => {
    const component = renderer.create(
        <MapScreen />
    )
    let tree = component.toJSON
    expect(tree).toMatchSnapshot();
})