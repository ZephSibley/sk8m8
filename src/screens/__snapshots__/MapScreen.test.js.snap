// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering 1`] = `
<div
  className="screen"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
  <div
    className="map-options"
  >
    <label
      className="MuiFormControlLabel-root makeStyles-standardSpacing-8"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          Log in / Sign up
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      />
    </label>
    <div
      className="MuiFormControl-root makeStyles-standardSpacing-8"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink={true}
        htmlFor="radius"
      >
        Radius
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <select
          aria-invalid={false}
          className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
          disabled={false}
          id="radius"
          name="radius"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          value={1}
        >
          <option
            value={0}
          >
            0
          </option>
          <option
            value={1}
          >
            1
          </option>
          <option
            value={5}
          >
            5
          </option>
          <option
            value={10}
          >
            10
          </option>
          <option
            value={25}
          >
            25
          </option>
          <option
            value={50}
          >
            50
          </option>
          <option
            value={100}
          >
            100
          </option>
        </select>
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiNativeSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
